### PR TITLE
Support launchtemplates

### PR DESCRIPTION
**Pull request by so0k**
Added support for LaunchTemplates

- Updates aws-sdk-go to add LaunchTemplate support
- Flags instances old when LaunchTemplate version of the instance does not match the LaunchTemplate version of the AutoScalingGroup

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/bouncer](https://github.com/palantir/bouncer), @so0k! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by so0k** [comment]
An attempt to implement https://github.com/palantir/bouncer/issues/31

File: bouncer/asg.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func NewASG(ac *aws.Clients, desASG *DesiredASG, force bool, startTime time.Time
 	var instances []*Instance
 
 	for _, asgInst := range awsAsg.Instances {
-		inst, err := NewInstance(ac, awsAsg, asgInst, awsAsg.LaunchConfigurationName, force, startTime, desASG.PreTerminateCmd)
+		inst, err := NewInstance(ac, awsAsg, asgInst, awsAsg.LaunchConfigurationName, awsAsg.LaunchTemplate, force, startTime, desASG.PreTerminateCmd)
```
Comment (so0k) [inline bouncer/asg.go]: awsAsg.LaunchConfigurationName and awsAsg.LaunchTemplate are already passed into the function through awsAsg itself, I don't see the point of having those additional function parameters actually... but I copied the same style

**Comment by holtwilkins** [comment]
Thanks for this @so0k , will make our eventual move to LTs much easier! We're fairly swamped at the moment, but hopefully I can take a look through this next week. Also, while I can view the two commits separately, given that they're not really related, how would you feel about breaking-out the AWS SDK update into its own PR, which I can then test on my side and merge independently?

**Comment by holtwilkins** [comment]
Hey @so0k , I'm going to go ahead and separate some of these things out a bit.

**Comment by holtwilkins** [comment]
Closing in favour of https://github.com/palantir/bouncer/pull/37/files

**Comment by so0k** [comment]
@holtwilkins - sorry for slow response

Thanks for taking this over and re-working it to fit your needs, I'm currently using a binary from my fork to do what I needed to day, it's great if the functionality is merged upstream and I can just use your version.

**Pull request closed without merging**